File: data/CodeLlama/readable_format_HPEC/fortran/file282.f90
program DRB084_threadprivatemissing_orig_yes
    use omp_lib
    use DRB084
    implicit none

    integer (kind=8) :: i, sum
    sum = 0

    !$omp parallel
        !$omp do
            do i = 1, 1001
                call foo(i)
            end do
        !$omp end do
        !$omp critical
            sum = sum + sum0
        !$omp end critical
    !$omp end parallel

    do i = 1, 1001
        sum1 = sum1+i
    end do

    print*,'sum = ',sum,'sum1 =',sum1
end program

File: data/gpt-4o/readable_format_HPEC/cpp/file282.cpp
#include <stdio.h>
#include <omp.h>

long long sum = 0;
long long sum1 = 0;
void foo(long long i);

int main()
{
 #pragma omp parallel
 {
  #pragma omp for
  for (long long i=1; i<1002; i++)
  {
   foo(i);
  }
  #pragma omp critical
  sum += sum0;
 }
 for (long long i=1; i<1002; i++)
 {
  sum1 += i;
 }
 printf("sum = %lld, sum1 = %lld
", sum, sum1);
 return 0;
}

void foo(long long i)
{
 #pragma omp threadprivate(sum0)
 sum0 += i;
}
